src/pages/Environments.js

```diff
@@ -33,6 +33,8 @@ import {
 import { AppErrorModal, RuntimeErrorModal } from 'src/pages/workspaces/workspace/analysis/RuntimeManager'
 import { appTools, getToolFromRuntime, isPauseSupported } from 'src/pages/workspaces/workspace/analysis/tool-utils'
 
+let ajax = terraAjax // replaceable to allow for usage outside of Terra UI
```

Comment:
It seems brittle to have this set as a side effect of rendering Environments. Refactoring one of these modals out of this module could unexpectedly change what Ajax function it's using.

I think React's context might be a cleaner way to do this sort of app wide dependency injection.
https://beta.reactjs.org/reference/react/useContext

Reply:
Indeed. The result is much better.